### PR TITLE
Fix #11 and adds optional message comparison

### LINKED ISSUE #11: expectToThrow should accept any Throwable
Only Types corresponding to subtypes of Exception can be passed to `com.athaydes.specks.asserions::expectToThrow(Type<Exception> expectedType)(Throwable? result)` in the left argument list, so I cannot test if a function throws, for example, an AssertionError.

### PR DESCRIPTION
**Pull request by dlkw**
Fixes #11.
Please note that I could not get the tests to pass on the original branch. Are they supposed to work?
I thus did not add a test for this change and couldn't observe if it breaks anything (but it shouldn't).

### CODE REVIEW

**Comment by renatoathaydes** [comment]
Thanks, I will look at it tonight!

File: source/com/athaydes/specks/assertion/assertions.ceylon [hunk 1]
```diff
@@ -43,16 +43,23 @@ shared AssertionResult expect<Element>(Element actual, Matcher<Element> matcher)
 shared AssertionResult expectCondition(Boolean expectedToBeTrue)
         => expect(expectedToBeTrue, toBe(identicalTo(true)));
 
-"Platform independent name of an Exception"
-shared String platformIndependentName(Type<Exception>|Throwable exception) =>
+"Platform independent name of a Throwable"
+shared String platformIndependentName(Type<Throwable>|Throwable exception) =>
         exception.string.replace("::", ".");
 
+"A value to signal \"no comparison should be done\" in an assertion"
```
Comment (renatoathaydes) [inline source/com/athaydes/specks/assertion/assertions.ceylon:50]: That's not how it's being used, it seems you just pass this value in to signal `expectToThrow`'s message should be not used??

File: source/com/athaydes/specks/assertion/assertions.ceylon [hunk 2]
```diff
@@ -43,16 +43,23 @@ shared AssertionResult expect<Element>(Element actual, Matcher<Element> matcher)
...
+ with exactly the [[expectedType]].
+ 
+ If the type check passes and parameter `message` is a String or `null`, the message of the `result` Throwable
+ must match that value for the assertion to be successful. The message is not checked if the `message` parameter is
+ set to `noCheck` (the default value).
 
  This assertion is commonly used with the [[errorCheck]] block."
 see(`function errorCheck`)
-shared AssertionResult expectToThrow(Type<Exception> expectedType)(Throwable? result) {
+shared AssertionResult expectToThrow(Type<Throwable> expectedType, String? | \InoCheck message = noCheck)(Throwable? result) {
```
Comment (renatoathaydes) [inline source/com/athaydes/specks/assertion/assertions.ceylon:62]: I don't understand how `noCheck` works... it doesn't seem to be used in any way?

**Review by renatoathaydes — COMMENTED** [review]
Please clarify `noCheck` and why it's being added.

**Comment by renatoathaydes** [comment]
To run the tests, run `./gradlew test`. They should all pass.

**Comment by renatoathaydes** [comment]
Thank you for your effort, but as this PR was incomplete and you've not answered my questions, I decided to fix the problem myself.
Please pull from master to get the latest fix (I will try to publish this later to the Herd).

**Comment by dlkw** [comment]
Sorry, I'm not able to work that fast due to real life pressure...
I just pushed the missing commit, if you'd like to take a look.
I think the optional message comparison is a useful addition.
Thanks anyway!

**Comment by renatoathaydes** [comment]
There's no new commits since 3 days ago.

**Comment by dlkw** [comment]
The pushes on my forked branch are obviously not visible here, maybe because they come from a forked repo or because this PR is closed. I'm preparing a new PR against current master.

**Pull request closed without merging**